src/cdomains/apron/sharedFunctions.apron.ml

```diff
@@ -75,9 +85,33 @@ struct
       | _ -> None (* for other exception *)
     )
 
-  let texpr1_expr_of_cil_exp d env exp no_ov =
+  (** This version with an overflow check is used by the apron domains such as polyhedra and octagons.
+      They do the overflow handling while they convert the expression to Texpr1. *)
+  let overflow_handling_apron no_ov ik env expr d exp =
+    if not (Lazy.force no_ov) then (
+      let (type_min, type_max) = IntDomain.Size.range ik in
+      let texpr1 = Texpr1.of_expr env expr in
+      match Bounds.bound_texpr d texpr1 with
+      | Some min, Some max when Z.compare type_min min <= 0 && Z.compare max type_max <= 0 -> ()
+      | min_opt, max_opt ->
+        if M.tracing then M.trace "apron" "may overflow: %a (%a, %a)\n" CilType.Exp.pretty exp (Pretty.docOpt (IntOps.BigIntOps.pretty ())) min_opt (Pretty.docOpt (IntOps.BigIntOps.pretty ())) max_opt;
+        raise (Unsupported_CilExp Overflow)
+    )
+
+  (** This version without an overflow check is used by the affeq and lin2vareq domains.
+          They do the overflow handling before converting to Texpr1, and store the result in the "no_ov" flag. Therefore we only check the no_ov flag here. *)
```

Comment:
The apron domains are not compatible with the new overflow checking method. They handle overflow differently than `affeq` and they rely on the bounds check that was made in `texpr1_expr_of_cil_exp` before I removed it. 
They cannot rely only on the `no_ov` flag to calculate the overflow, partly because they also store inequalities and because 
the `MaySignedOverflow` query is not suitable for finding overflow for unsigned expressions. 

But the code duplication is caused by the analysis of the casts in the expression, not the overflow analysis. The overflow analysis can be turned on or off by a flag set in the module `OverflowCheck`, therefore it doesn't cause code duplication.

Reply:
Overflow checking is unified now.